pkg/metastore/metastore.go

```diff
@@ -35,9 +56,19 @@ func (m *Partition) Update(db, table, serde string, transformLocation func(locat
 	m.TableName = table
 	m.Sd.SerdeInfo.Name = serde
 	var err error
-	if m.Sd.Location != "" {
+	if m.Sd.Location != "" && !(isSparkSQLTable && strings.HasSuffix(m.Sd.Location, sparkSQLWorkaroundSuffix)) {
 		m.Sd.Location, err = transformLocation(m.Sd.Location)
 	}
+	if err != nil {
+		return err
+	}
+
+	if isSparkSQLTable {
+		// Table was created by Spark SQL, we should change the internal stored Spark SQL location
+		if l, ok := m.Sd.SerdeInfo.Parameters[sparkSQLProviderLocationKey]; ok {
```

Comment:
note - the way we check and the way we update are the same in this case.  it is like we don't need to ask if `isSparkSQLTable` because if it was the next part will ask the same question and update.

Reply:
It's not the same, `isSparkSQLTable` checks for a spark provider
here we change the `path`, which can be used by other clients